spanner/client.go

```diff
@@ -134,6 +134,11 @@ type ClientConfig struct {
 	// override the default values.
 	CallOptions *vkit.CallOptions
 
+	// UserAgent is the prefix to the user agent header. This is used to supply information
+	// such as application name or partner tool.
+	// Recommended format: ``application-or-tool-ID/major.minor.version``.
+	UserAgent string
```

Comment:
This is now an exported field that could be used by anyone to set any random value. Would it be possible to either:
1. Somehow make this unexported? I know that it is called from a different package, so literally making it unexported without any other changes won't work. But would it be possible to in some way or another make this less accessible?
2. Or otherwise: At least mark this as an internal option that users should not set, and have a fixed list of values that are allowed to be set. Now anyone could set any value, and it does not in any way tell you that users should normally not do this.

Reply:
What will be the issue if user set custom values? since this will just append to the existing header 
example if userAgent is "go-sql-spanner/0.1" the header will look like 

```
gl-go/1.19.0 go-sql-spanner/0.1 gapic/1.36.0 gax/2.4.0 grpc/1.48.0
```

So no breaking changes to existing metrics and tracking, from driver and ORMs we will not expose this field for sure.

Reply:
In python also we are not adding any validation/check.
Example for Django we pass like this: https://github.com/googleapis/python-spanner-django/pull/140
And in the client lib there are no validations.

Reply:
Are you sure that this does not change anything? The handwritten Spanner client will normally also call `setClientInfo` with `gccl/<version>`. That seems to be overwritten if the application has set a user agent. At least, in your example above the `gccl/<version>` part seems to be missing, which means that it is not just appending.

And even then I would suggest that we at least document `UserAgent` as an internal field. There's nothing that users can do with it, and it might confuse them if they can set it as if it is the same as any other field, but they won't see it turn up anywhere.

Reply:
Make sense, please take a look